bin/search_solr_tools

```diff
@@ -22,10 +64,20 @@ class SolrHarvestCLI < Thor
       begin
         harvest_class = get_harvester_class(target)
         harvester = harvest_class.new(options[:environment], die_on_failure)
+        ping_status = SearchSolrTools::Helpers::HarvestStatus.new
+        ping_status.ping_solr = false unless harvester.ping_solr
+        ping_status.ping_source = false unless harvester.ping_source
+        raise SearchSolrTools::Errors::HarvestError, ping_status unless ping_status.ok?
+
         harvester.harvest_and_delete
+      rescue SearchSolrTools::Errors::HarvestError => e
+        puts "THERE WERE HARVEST STATUS ERRORS:\n#{e.message}"
+        exit e.exit_code
       rescue StandardError => e
+        # If it gets here, there is an error that we aren't expecting.
         puts "harvest failed for #{target}: #{e.message}"
-        raise e
+        puts e.backtrace
+        exit 99_999
```

Comment:
Wouldn't it be better to simply emit a non-zero integer code to indicate failure? What about using the ERRCODE_OTHER defined in the HarvestError class?

Reply:
Good idea.  My initial thought was to make this a different one, as this part is for unexpected errors, but "OTHER" would work just as well for that.